### ISSUE TITLE
Dialog cards should trap keyboard focus

### ISSUE THREAD

**Issue by svatas** [klembot/twinejs#1057]
### Describe the bug.

Handling of TAB key in GUI is wrong. If you press TAB key next UI element should be focused. Instead of this, the whole operationis not handled propelŕly as just a few elements is focused and then Editor window grab the TAB key and using it as for indentation

### Steps to reproduce:

1. Click to Story - New
2. Double click to Passage
3. Click to button Rename
4. Press TAB key, Edit line is focused
5. Press TAB key, button OK focused
6. Press TAB key, button Cancel focused
7. Press TAB key, cursor appears in Passage Edit window
8. Press TAB key, cursor will insert a Tab into Pssage Edit window

This is wrong.

### Expected behavior:

When button Cancel focused and TAB key pressed, the focus should now go to the Edit line again and then to OK, then to Cancel and then again to Edit line.

The same with SHIFT+TAB for opposite way.

### Twine version number

2.4.0 beta-2

### Does this problem occur with the web version of Twine or the desktop app?

Desktop app

### What operating system does this problem occur on?

Windows

### Presubmission checklist

- [ ] I am interested in working on code that would fix this bug. (This is not required to submit a bug report.)
- [X] I have done a search and believe that an issue does not already exist for this bug in the GitHub repository.
- [X] I have read and agree to abide by this project's Code of Conduct.

**Comment by klembot** [comment]
I think I understand what you're asking for, but I'm not sure this is correct. The dialogs are modeless--they do not require a response before the user can interact with other parts of the application.

I think I need to research this topic a little more to understand how this should be handled.
